### PR TITLE
Add option to save QgsProject locally when saving to postgres failed

### PR DESCRIPTION
**Pull request by PeterPetrik**
Add option to save QgsProject locally when saving to postgres failed

save_as_project

### CODE REVIEW

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -14096,11 +14096,21 @@ void QgisApp::populateProjectStorageMenu( QMenu *menu, bool saving )
...
           else
           {
-            QMessageBox::critical( this,
-                                   tr( "Unable to save project %1" ).arg( uri ),
-                                   QgsProject::instance()->error(),
-                                   QMessageBox::Ok,
-                                   Qt::NoButton );
+            QMessageBox msgbox;
+
+            msgbox.setWindowTitle( tr( "Unable to save project %1" ).arg( uri ) );
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp]: HIG convention is that message box title doesn't state the error, just the action. I.e. this should be "Save Project"

File: src/app/qgisapp.cpp [hunk 2]
```diff
@@ -14096,11 +14096,21 @@ void QgisApp::populateProjectStorageMenu( QMenu *menu, bool saving )
...
+            msgbox.setText( QgsProject::instance()->error() );
+            msgbox.setIcon( QMessageBox::Icon::Critical );
+            msgbox.addButton( QMessageBox::Cancel );
+            msgbox.addButton( QMessageBox::Save );
+            msgbox.setButtonText( QMessageBox::Save, tr( "Save as local file" ) );
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp]: Capitalization for button actions should be title case, "Save as Local File"

File: src/providers/postgres/qgspostgresprojectstorage.cpp [hunk 3]
```diff
@@ -162,7 +161,7 @@ bool QgsPostgresProjectStorage::writeProject( const QString &uri, QIODevice *dev
     QgsPostgresResult res( conn->PQexec( sql ) );
     if ( res.PQresultStatus() != PGRES_COMMAND_OK )
     {
-      QString errCause = QObject::tr( "Unable to save project. It's not possible to create the destination table on the database. Maybe this is due to table permissions (user=%1). Please contact your database admin" ).arg( projectUri.connInfo.username() );
+      QString errCause = QObject::tr( "Unable to save project. It's not possible to create the destination table on the database. Maybe this is due to database permissions (user=%1). Please contact your database admin" ).arg( projectUri.connInfo.username() );
```
Comment (nyalldawson) [inline src/providers/postgres/qgspostgresprojectstorage.cpp]: Missing "." after "Please contact your database admin"

File: src/providers/postgres/qgspostgresprojectstorage.cpp [hunk 4]
```diff
@@ -188,11 +187,16 @@ bool QgsPostgresProjectStorage::writeProject( const QString &uri, QIODevice *dev
   sql += "') ON CONFLICT (name) DO UPDATE SET content = EXCLUDED.content, metadata = EXCLUDED.metadata;";
 
   QgsPostgresResult res( conn->PQexec( sql ) );
-  bool ok = res.PQresultStatus() == PGRES_COMMAND_OK;
+  if ( res.PQresultStatus() != PGRES_COMMAND_OK )
+  {
+    QString errCause = QObject::tr( "Unable to insert or update project (project=%1) in the destination table on the database. Maybe this is due to table permissions (user=%2). Please contact your database admin" ).arg( projectUri.projectName, projectUri.connInfo.username() );
```
Comment (nyalldawson) [inline src/providers/postgres/qgspostgresprojectstorage.cpp]: same here

**Review by nyalldawson — COMMENTED** [review]
Code looks good to me

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**